view/frontend/templates/config.phtml

```diff
@@ -19,18 +19,22 @@
     <?php
     $frontName = $block->getRequest()->getFrontName();
     $pathInfo = $block->getRequest()->getPathInfo();
-    ?>
-<script>
+    $config = $block->getJsonConfig();
 
+    $script = <<<script
     require (['uiRegistry'], function(registry) {
-        registry.set('amazonPay', <?= /* @noEscape */ $block->getJsonConfig() ?>)
+        registry.set('amazonPay', {$config});
     });
+    script;
+    ?>
 
     <?php if ($frontName != 'checkout' || (strpos($pathInfo, 'checkout/cart') !== false)): ?>
-    require (['Amazon_Pay/js/model/storage'], function(amazonStorage) {
-        amazonStorage.clearAmazonCheckout();
-    });
+        <?php
+        $script .= PHP_EOL . "require (['Amazon_Pay/js/model/storage'], function(amazonStorage) {
+            amazonStorage.clearAmazonCheckout();
+        });";
+        ?>
     <?php endif; ?>
 
-</script>
+    <?= /* noEscape */ $csp->renderTag('script', [], $script) ?>
```

Comment:
What's `$csp` here? Could a docblock be added if it's a global variable?
Should we use the secureRenderer instead?

Reply:
It's a utility provided to the template engine by the CSP module: https://github.com/magento/magento2/blob/e81f28e4e6739924594c266b011c68b08a227c2d/app/code/Magento/Csp/etc/di.xml#L77-L86

I believe the renderTag function of that class uses a secureRenderer to actually add the content, but may have misunderstood. Docblock is a good idea but I'm not sure how/where it would be appropriate to add that

Reply:
Then it's fine to use it. I would just add the docblock at the top for the variable, in the same way as the `$block` variable